impeller/entity/shaders/blending/advanced_blend_softlight.frag

```diff
@@ -0,0 +1,18 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "advanced_blend_utils.glsl"
+
+vec3 Blend(vec3 dst, vec3 src) {
+  vec3 D = MixComponents(((16 * dst - 12) * dst + 4) * dst,  //
```

Comment:
I had to look up which variant you'd used on Wikipedia. I think it'd be good to document this is W3C variant here. Adding documentation for all blend modes either inline here or in a new doc in the `docs/` folder would be great (in a separate patch is fine).

Reply:
Yeah, all of these come from https://www.w3.org/TR/compositing-1/#blendingseparable. Added comments to each of these shaders linking the relevant sections. Documenting this along with the blend situation in general seems like a good idea -- I'll start working on a doc add it in a later patch.